std/typecons.d

```diff
@@ -3476,7 +3476,7 @@ Returns:
         }
         //Need to use 'is' if T is a float type
         //because NaN != NaN
-        else static if (isFloatingPoint!T)
+        else static if (__traits(isFloating, T) || __traits(compiles, { static assert(!(nullValue == nullValue)); }))
```

Comment:
Q: Why is there the seemingly-extraneous `__traits(isFloating, T)`?
A: So this patch doesn't change the behavior of `Nullable!(double, -0.0)`.